pushnotifications/src/main/java/com/pusher/pushnotifications/SubscriptionsChangedListener.kt

```diff
@@ -0,0 +1,13 @@
+package com.pusher.pushnotifications
+
+/**
+ * The listener interface for when a interests change upon migrating from a 3rd-party service
```

Comment:
I think we ought to make this listener fire whenever they change, and not necessarily just when a migration has occurred.

That way it's not a migration specific feature, but a stand-alone feature that could be used even if no migration is involved. E.g. My app subscribes and unsubscribes based on network requests with my backend server, and I want the UI to automatically update whenever this happens, so I can just set up this listener to achieve that.